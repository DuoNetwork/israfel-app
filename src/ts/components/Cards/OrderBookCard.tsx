import * as React from 'react';
import * as CST from 'ts/common/constants';
import { IOrderBook, IOrderBookSnapshot } from 'ts/common/types';
import util from 'ts/common/util';
import { SDivFlexCenter } from '../_styled';
import { SCard, SCardTitle } from './_styled';
import { SCardList } from './_styled';

interface IProps {
	OrderBookSnapshot: IOrderBookSnapshot;
}

interface IState {
	searchTextVolume: string;
	searchTextVolumeBidResult: string;
	searchTextVolumeAskResult: string;
	orderBook: IOrderBook;
}

let askArray: number[][] = [];
let bidArray: number[][] = [];

export default class OrderBookCard extends React.Component<IProps, IState> {
	constructor(props: any) {
		super(props);
		this.state = {
			searchTextVolume: '',
			searchTextVolumeBidResult: '',
			searchTextVolumeAskResult: '',
			orderBook: props.orderBook
		};
	}

	public render() {
		const { OrderBookSnapshot } = this.props;
		askArray = [];
		bidArray = [];
		if (OrderBookSnapshot) {
			const asks = OrderBookSnapshot.asks;
			const bids = OrderBookSnapshot.bids;
			for (let i = 0; i < asks.length; i++) askArray.push([asks[i].balance, asks[i].price]);
			for (let i = 0; i < bids.length; i++) bidArray.push([bids[i].balance, bids[i].price]);
		}
		askArray.sort((a, b) => a[1] - b[1]);
		bidArray.sort((a, b) => b[1] - a[1]);
		return (
			<div>
				<SCard
					title={<SCardTitle>{CST.DB_ORDER_BOOKS.toUpperCase()}</SCardTitle>}
					width="400px"
					margin="0 0 0 5px"
				>
					<SDivFlexCenter width="370px" center horizontal>
						<SCardList>
							<div className="status-list-wrapper">
								<ul>
									<li className="right">
										<span className="title">{CST.DB_BID.toUpperCase()}</span>
									</li>
									{bidArray.length ? (
										bidArray.map((bid, i) => (
											<li key={i}>
												<span className="content">
													{bid[1] !== 0
														? util.formatNumber(bidArray[i][0])
														: '-'}
												</span>
												<span className="title">
													{bid[0] !== 0
														? util.formatNumber(bidArray[i][1])
														: '-'}
												</span>
											</li>
										))
									) : (
										<li className="block-title t-center">-</li>
									)}
								</ul>
							</div>
						</SCardList>
						<SCardList>
							<div className="status-list-wrapper">
								<ul>
									<li>
										<span className="title">{CST.DB_ASK.toUpperCase()}</span>
									</li>
									{askArray.length ? (
										askArray.map((ask, i) => (
											<li key={i}>
												<span className="title">
													{ask[0] !== 0
														? util.formatNumber(askArray[i][1])
														: '-'}
												</span>
												<span className="content">
													{ask[1] !== 0
														? util.formatNumber(askArray[i][0])
														: '-'}
												</span>
											</li>
										))
									) : (
										<li className="block-title t-center">-</li>
									)}
								</ul>
							</div>
						</SCardList>
					</SDivFlexCenter>
				</SCard>
			</div>
		);
	}
}
