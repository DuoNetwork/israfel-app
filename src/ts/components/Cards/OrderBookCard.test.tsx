/**
 * @jest-environment jsdom
 */

import { shallow } from 'enzyme';
import * as React from 'react';
import OrderBookCard from './OrderBookCard';

describe('OrderBookCard Test', () => {
	describe('Test Snapshot', () => {
		const orderBook = {
			asks: [
				{price: 0.007, amount: 30, count: 3},
				{price: 0.007, amount: 4, count: 1}
			],
			bids: [
				{price: 0.008, amount: 0, count: 1},
				{price: 0.007, amount: 4, count: 1}
			],
			pair: "B-PPT-I0|WETH",
			version: 1543479830843
		};

		const orderBook1 = {
			asks: [],
			bids: [],
			pair: "B-PPT-I0|WETH",
			version: 1543479830843
		};

		it('Test Snapshot', () => {
			const wrapper = shallow(
				<OrderBookCard
					OrderBookSnapshot={orderBook}
				/>
			);
			expect(wrapper).toMatchSnapshot();
			const wrapper1 = shallow(
				<OrderBookCard
					OrderBookSnapshot={orderBook1}
				/>
			);
			expect(wrapper1).toMatchSnapshot();
			});
	});
});
