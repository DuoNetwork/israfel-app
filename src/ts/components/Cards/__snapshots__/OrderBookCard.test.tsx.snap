// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBookCard Test Test Snapshot Test Snapshot 1`] = `
<div>
  <Styled(Card)
    margin="0 0 0 5px"
    title={
      <styled.div>
        ORDERBOOKS
      </styled.div>
    }
    width="400px"
  >
    <styled.div
      center={true}
      horizontal={true}
      width="370px"
    >
      <styled.div>
        <div
          className="status-list-wrapper"
        >
          <ul>
            <li
              className="right"
            >
              <span
                className="title"
              >
                BID
              </span>
            </li>
            <li
              key="0"
            >
              <span
                className="content"
              >
                NANUNDEFINED
              </span>
              <span
                className="title"
              >
                0.008000
              </span>
            </li>
            <li
              key="1"
            >
              <span
                className="content"
              >
                NANUNDEFINED
              </span>
              <span
                className="title"
              >
                0.007000
              </span>
            </li>
          </ul>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="status-list-wrapper"
        >
          <ul>
            <li>
              <span
                className="title"
              >
                ASK
              </span>
            </li>
            <li
              key="0"
            >
              <span
                className="title"
              >
                0.007000
              </span>
              <span
                className="content"
              >
                NANUNDEFINED
              </span>
            </li>
            <li
              key="1"
            >
              <span
                className="title"
              >
                0.007000
              </span>
              <span
                className="content"
              >
                NANUNDEFINED
              </span>
            </li>
          </ul>
        </div>
      </styled.div>
    </styled.div>
  </Styled(Card)>
</div>
`;

exports[`OrderBookCard Test Test Snapshot Test Snapshot 2`] = `
<div>
  <Styled(Card)
    margin="0 0 0 5px"
    title={
      <styled.div>
        ORDERBOOKS
      </styled.div>
    }
    width="400px"
  >
    <styled.div
      center={true}
      horizontal={true}
      width="370px"
    >
      <styled.div>
        <div
          className="status-list-wrapper"
        >
          <ul>
            <li
              className="right"
            >
              <span
                className="title"
              >
                BID
              </span>
            </li>
            <li
              className="block-title t-center"
            >
              -
            </li>
          </ul>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="status-list-wrapper"
        >
          <ul>
            <li>
              <span
                className="title"
              >
                ASK
              </span>
            </li>
            <li
              className="block-title t-center"
            >
              -
            </li>
          </ul>
        </div>
      </styled.div>
    </styled.div>
  </Styled(Card)>
</div>
`;
